### DISCUSSION TITLE
Expanding DISTINCT to AVERAGE and SUM

### DISCUSSION THREAD

**Discussion by PhilTerz** [jamessimone/apex-rollup#583] [Ideas]
It would be so valuable to have an `AVERAGE_DISTINCT` and `SUM_DISTINCT` for rolling up child objects that may be referencing the same record. We have Contacts that are associated to different partitions in the system but rollup to the same great-grandparent. The inability to avoid double-counting is forcing us to write some metrics using custom work but really I would love to use Apex Rollup here.

**Comment by jamessimone** [comment]
@PhilTerz apologies for the delay in response - let me think on this one and get back to you. I don't foresee major blockers with implementing this, but there are a few things I'd like to consider before beginning work on it. I should have more info for you later in this week.

**Reply by jamessimone** [reply]
Still working on this feature, hope to have an update (and be done with it) this week

**Reply by PhilTerz** [reply]
Thanks so much! That's awesome

**Reply by jamessimone** [reply]
@PhilTerz this has now been released!
